post-til-virksomhet/src/main/java/no/difi/meldingsutveksling/ptv/CorrespondenceAgencyMessageFactory.java

```diff
@@ -13,147 +14,189 @@
-                binaryAttachmentV2.setName(reporteeFactory.createBinaryAttachmentV2Name(d.getFilename()));
-                binaryAttachmentV2.setEncrypted(false);
-                binaryAttachmentV2.setSendersReference(reporteeFactory.createBinaryAttachmentV2SendersReference("AttachmentReference_as123452"));
-                binaryAttachmentV2.setData(reporteeFactory.createBinaryAttachmentV2Data(Base64.getDecoder().decode(d.getContent())));
-                attachmentExternalBEV2List.getBinaryAttachmentV2().add(binaryAttachmentV2);
-            });
-
-            String title = PayloadUtil.queryPayload(edu.getPayload(), "Melding/journpost/jpInnhold");
-            String content = PayloadUtil.queryPayload(edu.getPayload(), "Melding/journpost/jpOffinnhold");
-
-            return create(config, edu.getId(), edu.getReceiver().getIdentifier(), title, content, attachmentExternalBEV2List);
-        } catch (PayloadException e) {
-            throw new MeldingsUtvekslingRuntimeException("Error querying payload for Dokument", e);
+            return cryptoMessagePersister.readStream(messageId, f.getIdentifier());
+        } catch (IOException e) {
+            throw new NextMoveRuntimeException(
+                    String.format("Could not read attachment %s for messageId = %s", f.getIdentifier(), messageId)
+                    , e);
         }
-
     }
 
-    public static InsertCorrespondenceV2 create(CorrespondenceAgencyConfiguration config,
-                                                String conversationId,
-                                                String receiverIdentifier,
-                                                String messageTitle,
-                                                String messageContent,
-                                                BinaryAttachmentExternalBEV2List attachments) {
+    public InsertCorrespondenceV2 create(String conversationId,
+                                         String receiverIdentifier,
+                                         String process,
+                                         String documentType,
+                                         String messageTitle,
+                                         String messageSummary,
+                                         String messageBody,
+                                         BinaryAttachmentExternalBEV2List attachments) {
 
         MyInsertCorrespondenceV2 correspondence = new MyInsertCorrespondenceV2();
         ObjectFactory objectFactory = new ObjectFactory();
 
+        ServiceRecord serviceRecord;
+        try {
+            serviceRecord = serviceRegistryLookup.getServiceRecord(
+                    SRParameter.builder(receiverIdentifier)
+                            .conversationId(conversationId).build(),
+                    process,
+                    documentType);
+        } catch (ServiceRegistryLookupException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)